### ISSUE TITLE
Documentation for WebScrapBook

### ISSUE THREAD

**Issue by Everest111** [danny0838/webscrapbook#113]
Is there any kind of documentation for WebScrapBook? We had a very helpful manual for the initial ScrapBook, which I have been using since 2008 (about 10,000 saved pages now). Probably not as detailed, but at least some documentation of the quick start-up steps would be very helpful.

Thank you.

**Comment by danny0838** [comment]
We have the [documentation wiki](https://github.com/danny0838/webscrapbook/wiki), whose link is provided in the README file and the descriotion in all browser stores.

**Comment by Everest111** [comment]
Thank you. Could you please send me a direct link to this documentation?

**Comment by danny0838** [comment]
I don't get you. Isn't there a link in my previous message?

**Comment by leggewie** [comment]
@danny0838 Thank you for this great tool and moving on with previous scrapbook extensions.  I am coming from Gomita's extension.

Thank you for the wiki and providing documentation there.  I have to admit that I haven't quite figured out how to use your extension yet.  I am using Chromium and Vivaldi on Ubuntu.  I am not sure the extension really does anything.  Are you testing with Chromium and/or Vivaldi

I haven't yet figured out how to get a list of captured pages and view them.  In the screenshots in the wiki, I see a side panel, but I have yet to find out how to trigger it.  Maybe that is something to add to the docu.

**Comment by Everest111** [comment]
In my opinion, ScrapBook could have been recognized by a much wider
population and even have become a very successful commercial product.
Internet is the main source of information nowadays, and having a tool to
properly store it in a highly organized form is invaluable. Unfortunately,
very few people know about this wonderful creation largely because there is
no sensible documentation on how to use it. There was a PDF manual for a
much older edition, but, it seems, there is nothing remotely close to it
for the new one. Even I, with my 10-year experience of enjoying previous
ScrapBook version and about 10,000 pages stored, cannot figure out how to
use the new version reliably and keep using the old one.

**Comment by leggewie** [comment]
Let's help @danny0838 to improve the documentation or even better make the extension so easy to use it doesn't need any docs.

**Comment by Everest111** [comment]
I would be glad to help, but, in order to express, one needs to understand.
:)

**Comment by leggewie** [comment]
which is why we have now both asked you if you had a look at the wiki, which is where documentation currently resides.

**Comment by Everest111** [comment]
Thanks for your message. Finally, I have got some time to look into it.
Independence Day will help. For any user starting to use this program, even
for those who have used ScrapBook and/or ScrapBook X, it is best to have,
first, some kind of a "Quick Start Guide", which would provide the very
basic functionality, namely:

- How to setup/configure the program in the simplest/fastest way;
- How to save a page;
- How to organize pages in forders/subfolders;
- How to search for a page (syntax of queries);
- How to open a page for reading or additional editing.

Once these things are mastered and the user is comfortable in his/her
belief that the pages are reliably saved and ready for use whenever needed,
he/she will be much more inclined to dive deeper.

I understand that, having matured for over a decade, the program has now
very rich functionality, and that is a very good thing. However, the
multitude of options from the start are pretty daunting, especially for
busy people who do not have a lot of time for experimenting.

So, in the next few days, I will go through the documentation in Wiki
again. Maybe, I will have some questions. Having gone through this, I will
have a clearer idea what actually is needed as a manual. I liked the manual
written for the initial ScrapBook by some Dutch, if I remember correctly,
and something like this would be valuable for this new version. However,
right now we need a Quick Start Guide for broad audience.

Cheers,

Vitaly

**Comment by Everest111** [comment]
Is there a way to communicate with Danny directly, like by e-mail? I am
trying to create a "Quick Start Guide" mentioned previously, and I have
several questions that would be best resolved in a direct communication. My
e-mail is vitaly.nagy@gmail.com.

Thank you.

**Comment by Michaelsy** [comment]
> that would be best resolved in a direct communication

If nothing speaks against, I would like to read along, maybe in a new thread?

**Comment by Everest111** [comment]
I am not against either option. On the one hand, interested people can be
added to e-mail communications as well. On the other hand, I would think
that answering my questions in public would be helpful for other people
stumbling over the same problems. If through the site, should we start a
new thread? What do you think, Danny?

**Comment by danny0838** [comment]
@Everest111 You can post in this issue as long as you stick to this topic. For example, ask a question that's already written in the documentation but you still aren't clear about. Or provide an idea how the documentation can be improved.

**Comment by Everest111** [comment]
I have performed installation as described on the "pip install webscrapbook" page. Everything went smoothly. Then, I needed to import my old ScrapBook X data. I had checked their health prior to the attempts to import ("Good"). As I mentioned before, I have some 10,000 saved pages totalling  734,000+ files of the total size about 20 GB. According to the instructions on the "Import ScrapBook X Data", I copied the data into a new location (did not delete anything), entered the indexer using the "Generate Site Index" option of the toolbar button and dragged the folder with the copied data into the tab of the indexer. As can be seen from the attached file of screenshots, initially everything seemed to run more or less normally. However, somewhere down the road, apparently at the stage of creating caches of the saved pages, the operation abruptly stopped. In various attempts to repeat the procedure, I got messages "Web ScrapBook has crashed", "Not enough memory to display this page" (or something like that) or the friendly "Aw, Snap!" message shown in the attached file. This is a fairly powerful machine with a 16-core AMD Threadripper-Pro CPU (7% max of whose capacity was used in the process), 128 G of RAM and dozens of TB of disk space. I use Windows 10 Pro with all the latests updates and patches. So, what should I do now? I would not mind if Danny connects to the machine remotely and tries to examine/resolve the problem with his skillful hands.  My e-mail is given above. Thanks.

[WebScrapBook Screenshots.docx](https://github.com/danny0838/webscrapbook/files/3513895/WebScrapBook.Screenshots.docx)

**Comment by danny0838** [comment]
@Everest111 What do you mean by closing the issue? Is the issue resolved?

If not, try unchecking "update fulltext cache" in options and report if it works or not.

**Comment by Everest111** [comment]
The issue is indeed not resolved. I will try to uncheck "update fulltext
cache" in Options this night.

Thank you.

**Comment by Everest111** [comment]
With the checkbox unchecked, everything has worked fine. Thank you. I have a tree visible from tree\map.html, and clicking on an item in it opens the page. Should I check the checkbox back on?

Now, how can I build on this set of data? 

How can I instruct the system to save new pages in the same set of directories? I have them in
P:\Internet\Web Content\Scrapbook Data\
How can I display the tree of the folders in the left panel in Chrome?
Will I be able to move items from one subfolder to another? 
Will I get the familiar property sheet before saving a page in order to select the folder and put keywords in the Comment field like I did before?

It appears that some additional configuration in Options is needed. I put http://localhost:8080/ as the address of the backend server, but there are also fields for User and Password. What are they?

I am attaching a file with screenshots of Options in their present state and output that I got in a attempt to save a page. What should be changed in the system to create the familiar environment of Scrapbook X as much as possible?

Thank you.
[WebScrapbook Screenshots 2019_08_19.docx](https://github.com/danny0838/webscrapbook/files/3518157/WebScrapbook.Screenshots.2019_08_19.docx)

**Comment by danny0838** [comment]
> With the checkbox unchecked, everything has worked fine. Thank you. I have a tree visible from tree\map.html, and clicking on an item in it opens the page. Should I check the checkbox back on?

The issue seems to be that Chrome runs out of memory when trying to update the fulltext cache, and thus you are likely to get the same result if you just provide same data to Chrome.

You can however try using another browser such as Firefox to do the "import". As it uses a different memory management strategy from Chrome you are this possible to get a different result.

> 
> Now, how can I build on this set of data?
> 
> How can I instruct the system to save new pages in the same set of directories? I have them in
> P:\Internet\Web Content\Scrapbook Data

It depends on where and how you run the backend server. If you run the backend server at `C:\Users\YourName`, new captures will be saved to somewhere under `C:\Users\YourName`, depending on your server configurations (consult `wsb config --help` for how to edit configs and `wsb help config` for details).

> How can I display the tree of the folders in the left panel in Chrome?

Just run `Open ScrapBook` in the dropdown list of the toolbar button after you have configured your server properly in the options. Note that Chrome does not support sidebars and the "sidebar" will be shown in another window instead.

> Will I be able to move items from one subfolder to another?

You can do this via the sidebar.

> Will I get the familiar property sheet before saving a page in order to select the folder and put keywords in the Comment field like I did before?

This is currently not implemented as it complicates things and does not work well when the user captures multiple pages at once. We are considering whether and how to implement it in the future. As for now, simply do this  via the sidebar after capture completes.

> 
> It appears that some additional configuration in Options is needed. I put http://localhost:8080/ as the address of the backend server, but there are also fields for User and Password. What are they?

They are empty by default and can be configured from the backend server (consult `wsb config --help` and `wsb help config`, again). You probably don't need that unless you are trying to expose your server to the whole world.

> 
> I am attaching a file with screenshots of Options in their present state and output that I got in a attempt to save a page. What should be changed in the system to create the familiar environment of Scrapbook X as much as possible?

I really don't get you. What is the relationship between WebScrapBook and the file you mentioned? Please be more detailed.

> Thank you.
> [WebScrapbook Screenshots 2019_08_19.docx](https://github.com/danny0838/webscrapbook/files/3518157/WebScrapbook.Screenshots.2019_08_19.docx)

**Comment by Everest111** [comment]
Danny,

Thank you very much for your reply.

Let me start with the most important thing for me. As I wrote before, I
used the Comment field in the metadata primarily to store keywords for
subsequent search. As you know, ScrapBook X allowed for instant search for
words in that field without reindexing. Apparently, it was just an instring
search through all the comments combined. That capability is extremely
important for me. With the number of stored pages that I have, a full-text
search, even with an index, is pretty much useless: the resulted queries
are huge.  On the other hand, once you understand how the search engine
works and compose the contents of the Comment fields according, you get an
extremely effective and efficient search tool: pretty much everything in
the huge collection is at your fingertips just a few keystrokes away.

Does Web Scrapbook have anything like that? If so, please instruct me how
to use it. If not, then I should probably wait with my switch from
ScrapBook X to WebScrapBook until this functionality is available.

If I am in the latter unfortunate situation, may I ask you for the old
Firefox extension for ScrapBook X? I keep an installer of a
pre-revolutionary version of Firefox (is it Version 56?) specifically for
ScrapBook X. I do not use it for anything else, and I do not allow for an
automatic update. Not an ideal situation, but better than not to use my
huge collection of stored web pages at all. However, understandably, the
extension is not available at the Firefox add-on site anymore, although you
almost certainly have it. There should still be some way to install it if I
get the installer from you directly. Right now, ScrapBook X with in old
Firefox version runs on an old computer, which needs to be replaced.

I would very much like to use the new WebScrapBook, but, if it cannot yet
provide the functionality critically important for me, it would be better
to use the previous version of your really amazing program than not to use
it at all.

Thank you.

**Comment by Michaelsy** [comment]
> may I ask you for the old Firefox extension for ScrapBook X?

[Here you can download the last version of the XPI file.](https://github.com/danny0838/firefox-scrapbook/releases) (Maybe you have to click on "Assets")

HTH - Michaelsy

**Comment by Everest111** [comment]
Thank you. I have downloaded the XPI file, but my Firefox 56.0.2 refuses to install it because it is unsigned. How can I bypass that while still keeping Firefox at Version 56.0.2?

**Comment by Michaelsy** [comment]
Take a look at http://mzl.la/1J7Lcsp :

> What are my options if I want to use an unsigned add-on? (advanced users)
>
> Firefox Extended Support Release (ESR), Firefox Developer Edition and Nightly versions of Firefox will allow you to override the setting to enforce the extension signing requirement, by changing the preference xpinstall.signatures.required to false in the Firefox Configuration Editor (about:config page). There are also special unbranded versions of Firefox that allow this override. See the MozillaWiki article, Add-ons/Extension Signing for more information. 

I'm using FF 52.9.0 ESR. In terms of security, this version may even be more recent than 56.0.2 because it is an ESR version.

**Comment by Everest111** [comment]
Thank you, I had read that. But where can I get an installer for FF 52.9.0
ESR, or any ESR for Version 52? It is ancient past...

**Comment by Michaelsy** [comment]
> where can I get an installer

https://ftp.mozilla.org/pub/firefox/releases/

**Comment by danny0838** [comment]
> Danny,
> 
> Thank you very much for your reply.
> 
> Let me start with the most important thing for me. As I wrote before, I
> used the Comment field in the metadata primarily to store keywords for
> subsequent search. As you know, ScrapBook X allowed for instant search for
> words in that field without reindexing. Apparently, it was just an instring
> search through all the comments combined. That capability is extremely
> important for me. With the number of stored pages that I have, a full-text
> search, even with an index, is pretty much useless: the resulted queries
> are huge.  On the other hand, once you understand how the search engine
> works and compose the contents of the Comment fields according, you get an
> extremely effective and efficient search tool: pretty much everything in
> the huge collection is at your fingertips just a few keystrokes away.
> 
> Does Web Scrapbook have anything like that? If so, please instruct me how
> to use it. If not, then I should probably wait with my switch from
> ScrapBook X to WebScrapBook until this functionality is available.
> 
Yes, WebScrapBook supports comments searching. You can enter the search page `search.html` (or click the search button from the `map.html`) and test it. This is independent to the fulltext cache and can thus work without it. The search syntax is similar to what [ScrapBook X uses](https://github.com/danny0838/firefox-scrapbook/wiki/Features#search-data), although we may change it in the future.

> If I am in the latter unfortunate situation, may I ask you for the old
> Firefox extension for ScrapBook X? I keep an installer of a
> pre-revolutionary version of Firefox (is it Version 56?) specifically for
> ScrapBook X. I do not use it for anything else, and I do not allow for an
> automatic update. Not an ideal situation, but better than not to use my
> huge collection of stored web pages at all. However, understandably, the
> extension is not available at the Firefox add-on site anymore, although you
> almost certainly have it. There should still be some way to install it if I
> get the installer from you directly. Right now, ScrapBook X with in old
> Firefox version runs on an old computer, which needs to be replaced.

As Michaelsy has mentioned, you can download them in the [release list of ScrapBook X](https://github.com/danny0838/firefox-scrapbook/releases). But [there are something you need to take care about](https://github.com/danny0838/firefox-scrapbook#installation) as Mozilla has stopped supporting legacy add-ons.

> 
> I would very much like to use the new WebScrapBook, but, if it cannot yet
> provide the functionality critically important for me, it would be better
> to use the previous version of your really amazing program than not to use
> it at all.
> 
> Thank you.

**Comment by Everest111** [comment]
Thank you. But how can I put in keywords for new pages?

**Comment by danny0838** [comment]
> 
> Thank you. But how can I put in keywords for new pages?
Via the sidebar. Click on an item to highlight it, and then use the dropdown command to edit the metadata.

**Comment by Everest111** [comment]
Many thanks to everybody who participated in this discussion and gave me
valuable information. It was helpful. Upon consideration of many factors, I
have decided that, for now, I will stick with the old "ScrapBook X" because
its interface appears to be smoother. Obviously, I will have to go to Web
Scrapbook at some point. If nothing else, it is ridiculous to have a
separate outdated browser solely for ScrapBook, but I would like to make a
switch when WebScarapBook gets a little more mature. As an answer to an
earlier question about how the documentation for WebScrapBook can be
improved, I am attaching a file with a tutorial for the very first version
of ScrapBook written more than 10 years ago. If the goal is wide usage of
this very valuable, but largely overlooked program, there is a need for
something like that. If it is not written by the time when I move to
WebScrapBook, I will be willing to write it.

Many thanks again to everybody.

**Comment by danny0838** [comment]
> 
> Many thanks to everybody who participated in this discussion and gave me valuable information. It was helpful. Upon consideration of many factors, I have decided that, for now, I will stick with the old "ScrapBook X" because its interface appears to be smoother. Obviously, I will have to go to Web Scrapbook at some point. If nothing else, it is ridiculous to have a separate outdated browser solely for ScrapBook, but I would like to make a switch when WebScarapBook gets a little more mature. As an answer to an earlier question about how the documentation for WebScrapBook can be improved, I am attaching a file with a tutorial for the very first version of ScrapBook written more than 10 years ago. If the goal is wide usage of this very valuable, but largely overlooked program, there is a need for something like that. If it is not written by the time when I move to WebScrapBook, I will be willing to write it. Many thanks again to everybody.

Thank you. Unfortunately I cannot see the file. GitHub doesn't support uploading files via email attachment. You may have to upload it via the web interface.

**Comment by Everest111** [comment]
Attached is the tutorial mentioned in my previous message. It is solely due to this document that I started using ScrapBook eleven years ago (I am so glad that I did: now I have some 10,000 pages stored in a highly organized form). Without such a tutorial, I would have certainly passed by this extremely valuable tool, as, I am sure, do many people with WebScrapBook now. Many thanks to Danny for this creation and its persistent development through the stormy period of the Firefox revolution!

[ScrapbookTutorial-1.2.pdf](https://github.com/danny0838/webscrapbook/files/3540781/ScrapbookTutorial-1.2.pdf)
